### PR TITLE
Add type ascription in play.api.test.Fakes.scala, and remove useless method

### PR DESCRIPTION
**Pull request by 3tty0n**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you [squashed your commits]? (Optional, but makes merge messages nicer.)
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

## Purpose

This adds type ascription to `path` and `withMultipartFormDateBody` method  in `play.api.test.FakeRequest` and `play.api.test.RequestTarget`. 
And remove useless method.

### CODE REVIEW

File: framework/src/play-test/src/main/scala/play/api/test/Fakes.scala [hunk 1]
```diff
@@ -112,7 +112,7 @@ class FakeRequest[A](request: Request[A]) extends Request[A] {
     withBody(body = AnyContentAsFormUrlEncoded(play.utils.OrderPreserving.groupBy(data.toSeq)(_._1)))
   }
 
-  def certs = Future.successful(IndexedSeq.empty)
+  def certs: Future[IndexedSeq[Nothing]] = Future.successful(IndexedSeq.empty)
```
Comment (gmethvin) [inline framework/src/play-test/src/main/scala/play/api/test/Fakes.scala]: I actually have no idea where this method is used. I would assume it's meant to be a `Future[Seq[X509Certificate]]` but if it's not needed anywhere maybe we should just remove it.

/cc @playframework/core

Comment (3tty0n) [inline framework/src/play-test/src/main/scala/play/api/test/Fakes.scala]: @gmethvin  Thank you for replying. Temporally should I fix the type ascription of certs `Future[IndexSeq[Nothing]]` to `Future[Seq[X509Certificate]]` ?

Comment (gmethvin) [inline framework/src/play-test/src/main/scala/play/api/test/Fakes.scala]: @3tty0n After some investigation it seems the method was added accidentally in [an unrelated commit](https://github.com/playframework/playframework/commit/b291def0f12dfe9539646cdca2e0357d7fc2cc5e#diff-a873d3f9c703672fb9415cd547f37fb7R101). It was never documented.

Since the method is essentially useless, I suggest we remove it completely.

Comment (3tty0n) [inline framework/src/play-test/src/main/scala/play/api/test/Fakes.scala]: @gmethvim OK, I remove `certs` in `FakeRequest` completely in this PR. May I change the title of this PR to such as `Remove useless method in FakeRequest and ...` ?
